apps/server/src/api-integration/integration.utils.ts

```diff
@@ -6,19 +6,40 @@ import { editEvent } from '../services/rundown-service/RundownService.js';
 import { getEventWithId } from '../services/rundown-service/rundownUtils.js';
 import { coerceBoolean, coerceColour, coerceNumber, coerceString } from '../utils/coerceType.js';
 
-const whitelistedPayload = {
+/**
+ *
+ * @param {number} value time amount in seconds
+ * @returns {number} time in milliseconds clamped to 0 and max duration
+ */
+function clampToMaxDuration(value: number) {
+  return Math.max(0, Math.min(value * MILLIS_PER_SECOND, maxDuration));
+}
+
+let rateLimitTimeout: NodeJS.Timeout | null = null;
+
+function rateLimit(value: unknown) {
+  if (rateLimitTimeout) {
+    throw new Error('time endport rate limited');
+  }
+  rateLimitTimeout = setTimeout(() => (rateLimitTimeout = null), 1000);
+  return value;
+}
```

Comment:
something like this?

Reply:
could we use one of our debounce function?
We want our rate limiting to be much smaller, like 20 calls per second or something

Reply:
I don't think the debounce or throttle dose what we want, it looks to they would potently set the value with a delay
here I think we should do it immediately or drop it, not buffer it?

I would think it is good to send a rejected message back so the call knows why it did not work?

Reply:
I might be misunderstanding, but I believe your assumptions are not correct

I don't think the debounce or throttle dose what w

> e want, it looks to they would potently set the value with a delay
> here I think we should do it immediately or drop it, not  buffer it?

Debounce does not buffer function calls, it will always ensure that the function is eventually called, while maintaining a call rate. which means it may drop intermediate values
I am not sure if our debounce function fits the bill
https://css-tricks.com/debouncing-throttling-explained-examples/

> I would think it is good to send a rejected message back so the call knows why it did not work?

I would generally agree but in this case i feel its unnecessary and we are just contributing to the network noise

Lets say the user does something weird in touchdesigner and starts pushing data to ontime at 60fps.
we would activate our rate limit and ignore all the changes that do not fit into our (eg) 20 calls per second.
In this case, the user would still see ontime update, we internally just decided that there is no point updating so often. In the same light, there is no point sending 40 error messages per second to the client saying that the change did not apply while, in fact, the user can see the applied value